jquery-ajax-localstorage-cache.js

```diff
@@ -7,6 +7,13 @@
      * or one generated from the url, the type and, if present, the data.
      */
     var genCacheKey = function (options) {
+        //in case cachey key is required to be generated dynamically by a function
+        if(
+            options.cacheKey && 
+            typeof(options.cacheKey) == typeof(function(){})
```

Comment:
No need for the expense of declaring a no-op function here. Instead, use:
`typeof options.cacheKey === 'function'`